hocuspocus-extension-elasticsearch/src/__test__/elasticsearch.test.ts

```diff
@@ -0,0 +1,86 @@
+import { newHocuspocus, newHocuspocusProvider, delayForMs } from 'test/utils';
+import { Elasticsearch } from '../elasticsearch';
+import elasticsearch from '@elastic/elasticsearch';
+
+describe('elasticsearch extension', () => {
+  let server;
+  const elasticsearchOpts: elasticsearch.ClientOptions = {
+    node: process.env.ELASTICSEARCH_URL,
+  };
+  beforeAll(async () => {
+    server = await newHocuspocus({
+      yDocOptions: { gc: false, gcFilter: () => true },
+      port: process.env.PORT ? Number(process.env.PORT) : 1234,
+
+      extensions: [
+        new Elasticsearch({
+          elasticsearchOpts,
+        }),
+      ],
+    });
+  });
+  afterAll(async () => {
+    await server.destroy();
+    await delayForMs(1000);
+  });
+
+  it('return default ydoc when fetched documentName does not exist', (resolve) => {
+    const provider = newHocuspocusProvider(server, {
+      onSynced() {
+        expect(provider.document.share.size).toBe(0);
+        provider.disconnect();
+        provider.destroy();
+        resolve();
+      },
+    });
+  });
+
+  it('return fetched ydoc', async () => {
+    const textName = 'test_name';
+
+    const provider1 = newHocuspocusProvider(server, {
+      onSynced() {
+        const ydoc = provider1.document;
+        ydoc.getText(textName).insert(0, 'foo');
+        provider1.disconnect();
+        provider1.destroy();
+      },
+    });
+
+    // wait for provider1 to connect and close
+    await delayForMs(1000);
+
+    const provider2 = newHocuspocusProvider(server, {
+      onSynced() {
+        expect(provider2.document.getText(textName)).toMatchInlineSnapshot(
+          `"foo"`
+        );
+        provider2.disconnect();
+        provider2.destroy();
+      },
+    });
+    // wait for provider2 to connect and close
+    await delayForMs(1000);
+  });
+
+  it('logs error', (resolve) => {
```

Comment:
Question: do we assert anything in this test?

Reply:
No, it seems that we can't catch extension error in hocuspocus server, so I just log error instead of throwing it. This test is just run the `catch` part to increase coverage.

Reply:
Got it. Maybe consider `jest.spyOn` console.log / console.error and expect console is called?

Another alternative would be just [adding `/* istanbul ignore next */` on error handlers](https://github.com/cofacts/rumors-line-bot/pull/376/files#diff-3d74613acb1d956e12832d77da2626ce51445b2b2ced75beccd03847b125f878), if the error does not happen that frequently.